configure.ac

```diff
@@ -652,21 +717,36 @@ if test "x$enable_md2" = "xyes"; then
 	enable_md2=yes
 else
 	enable_md2=no
-	CFLAGS="$CFLAGS -DNOMD2"
+	AC_DEFINE([NOMD2])
 fi
 AM_CONDITIONAL([ENABLE_MD2], [test "x$enable_md2" = "xyes"])
 
 AC_SUBST([pkcsslotd_user])
 AC_SUBST([pkcs_group])
+AC_DEFINE([PKCS64])
+AC_DEFINE([_XOPEN_SOURCE], [600], [POSIX 2004])
+AC_PROG_CC([xlc_r xlc gcc clang])
+AC_PROG_CC_STDC
+if test "x$build_aix" = "xyes"; then
+	CFLAGS="$CFLAGS -qinfo=pro:cns:lan:nocmp:nocnv:nogot:notrd -qalign=natural -qarch=pwr7 -qsuppress=1506-1527:1500-029 -qlanglvl=stdc99 -qsrcmsg -lbsd -q64"
+	# msg is used to ensure compilers that ignore the error pragma are still caught
+	AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
+		[[#if (__xlC__ >> 8) < 16
+			#msg "xlc 16.x or newer required"
+			#endif]]
+		)],,
+		[AC_MSG_ERROR([xlc 16.x or newer is required. Please install a newer release of the xlc compiler.])])
+else
+	CFLAGS="$CFLAGS -Wall -Wextra"
+	CFLAGS+=" -std=c99 -Werror=implicit-int -Werror=implicit-function-declaration -Werror=int-conversion -Werror=strict-prototypes -Werror=old-style-definition -pedantic"
+fi
 
-CFLAGS="$CFLAGS -DPKCS64 -D_XOPEN_SOURCE=600 -Wall -Wextra"
+AC_DEFINE_UNQUOTED([PKCSSLOTD_USER], "$pkcsslotd_user")
+AC_DEFINE_UNQUOTED([PKCS_GROUP], "$pkcs_group")
 
-CFLAGS+=' -DCONFIG_PATH=\"$(localstatedir)/lib/opencryptoki\" -DSBIN_PATH=\"$(sbindir)\" -DLIB_PATH=\"$(libdir)\" -DLOCKDIR_PATH=\"$(lockdir)\" -DOCK_CONFDIR=\"$(sysconfdir)/opencryptoki\" -DOCK_LOGDIR=\"$(logdir)\"'
-CFLAGS+=' -DPKCSSLOTD_USER=\"$(pkcsslotd_user)\" -DPKCS_GROUP=\"$(pkcs_group)\"'
-CFLAGS+=" -std=c99 -Werror=implicit-int -Werror=implicit-function-declaration -Werror=int-conversion -Werror=strict-prototypes -Werror=old-style-definition -pedantic"
+CFLAGS="$CFLAGS"' -DCONFIG_PATH=\"$(localstatedir)/lib/opencryptoki\" -DSBIN_PATH=\"$(sbindir)\" -DLIB_PATH=\"$(libdir)\" -DRUN_PATH=\"$(RUN_PATH)\" -DLOCKDIR_PATH=\"$(lockdir)\" -DOCK_CONFDIR=\"$(sysconfdir)/opencryptoki\" -DOCK_LOGDIR=\"$(logdir)\"'
 
 # At this point, CFLAGS is set to something sensible
-AC_PROG_CC
 AX_PROG_CC_FOR_BUILD
 AC_PROG_CXX
```

Comment:
Don't you need to supply a search list for C++ as well ? I.e. to prefer AIX's xls compiler over gcc ?

Reply:
Is this still valid? I don't see C++ being used anywhere during the build process. The language is popped off after the test, so the only language that should be set is just C at that point.

Reply:
Yes, we have one test (https://github.com/opencryptoki/opencryptoki/blob/master/testcases/build/cpp_test.cpp) that checks that the public header files can be used within a C++ program.

Other than that, we don't need C++ for opencryptoki itself.